src/G11n/Currency.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\G11n;
+
+use App\Exceptions\InvalidCurrencyException;
+
+final class Currency
+{
+    private const ALLOWED_VALUES = [
+        'EUR',
+        'USD',
+    ];
+    private $value;
+
+    public function __construct(array $currency)
+    {
+        foreach ($currency as $val) {
```

Comment:
Currency should be a string, so no foreach.
Also, if you're assigning to `$this->value` each time it's basically as if you do it only for the last case.